executor/insert_common.go

```diff
@@ -1034,3 +1051,88 @@ func (e *InsertValues) addRecordWithAutoIDHint(ctx context.Context, row []types.
 	}
 	return nil
 }
+
+type insertRuntimeStat struct {
+	*execdetails.BasicRuntimeStats
+	*tikv.SnapshotRuntimeStats
+	checkInsertTime time.Duration
+	rpcTime         time.Duration
+}
+
+func (e *insertRuntimeStat) String() string {
+	var prepareStr, rpcStatsStr, checkInsertStr, insertStr string
+	if e.checkInsertTime != 0 {
+		prepareStr = fmt.Sprintf("prepare:%v, ", time.Duration(e.BasicRuntimeStats.GetTime())-e.checkInsertTime)
+		if e.rpcTime != 0 {
+			checkInsertStr = fmt.Sprintf("check_insert:{total_time:%v, mem_check_insert:%v, prefetch:%v, rpc:{", e.checkInsertTime, e.checkInsertTime-e.rpcTime, e.rpcTime)
+		} else {
+			insertStr = fmt.Sprintf("insert:%v", e.checkInsertTime)
+		}
+	}
+	if e.SnapshotRuntimeStats != nil {
+		rpcStatsStr = e.SnapshotRuntimeStats.String()
+	}
+	var result string
+	if prepareStr != "" {
+		result += prepareStr
+	}
+	if checkInsertStr != "" {
+		result += checkInsertStr
+	}
+	if insertStr != "" {
+		result += insertStr
+	}
+	if rpcStatsStr != "" {
+		result += rpcStatsStr
+	}
+	if checkInsertStr != "" {
+		return result + "}}"
+	}
+	return result
+}
+
+// Clone implements the RuntimeStats interface.
+func (e *insertRuntimeStat) Clone() execdetails.RuntimeStats {
+	newRs := &insertRuntimeStat{}
+	if e.SnapshotRuntimeStats != nil {
+		snapshotStats := e.SnapshotRuntimeStats.Clone()
+		newRs.SnapshotRuntimeStats = snapshotStats.(*tikv.SnapshotRuntimeStats)
+	}
+	if e.rpcTime != 0 {
+		newRs.rpcTime = e.rpcTime
+	}
+	if e.checkInsertTime != 0 {
+		newRs.checkInsertTime = e.checkInsertTime
+	}
+	return newRs
+}
+
+// Merge implements the RuntimeStats interface.
+func (e *insertRuntimeStat) Merge(other execdetails.RuntimeStats) {
+	tmp, ok := other.(*insertRuntimeStat)
+	if !ok {
+		return
+	}
+	if tmp.SnapshotRuntimeStats != nil {
+		if e.SnapshotRuntimeStats == nil {
+			snapshotStats := tmp.SnapshotRuntimeStats.Clone()
+			e.SnapshotRuntimeStats = snapshotStats.(*tikv.SnapshotRuntimeStats)
+		} else {
+			e.SnapshotRuntimeStats.Merge(tmp.SnapshotRuntimeStats)
+		}
+	}
+	e.rpcTime += tmp.rpcTime
+	e.checkInsertTime += tmp.checkInsertTime
+}
+
+// Tp implements the RuntimeStats interface.
+func (e *insertRuntimeStat) Tp() int {
+	return execdetails.TpInsertRuntimeStat
+}
+
+func (e *runtimeStatsWithSnapshot) String() string {
+	if e.SnapshotRuntimeStats != nil {
+		return e.SnapshotRuntimeStats.String()
+	}
+	return ""
+}
```

Comment:
Move to `point_get.go` and put together with `runtimeStatsWithSnapshot ` definition.